packages/origin-backend/src/pods/user/user.controller.ts

```diff
@@ -73,4 +75,22 @@ export class UserController {
     public async get(@Param('id') id: string) {
         return this.userService.findById(id);
     }
+
+    @Put('profile/:id')
+    @UseGuards(AuthGuard('jwt'))
+    public async putProfile(@Param('id') id: string, @Body() body: IUser) {
+        return this.userService.updateProfile(id, body);
+    }
+
+    @Put('password')
+    @UseGuards(AuthGuard('jwt'))
+    public async putChainAddress(@Body() body: UserPasswordUpdate) {
+        return this.userService.updatePassword(body);
+    }
+
+    @Put('chainAddress/:id')
+    @UseGuards(AuthGuard('jwt'))
+    public async putPassword(@Param('id') id: string, @Body() body: IUser) {
+        return this.userService.updateBlockChainAddress(id, body);
```

Comment:
Method called `putPassword()` but updates blockchain address.

Also missing proper auth. The user should only be able to update his own blockchain account address.